apps/consumer-client/serve.ts

```diff
@@ -0,0 +1,39 @@
+import fs from "node:fs";
+import http from "node:http";
+
+// esbuild's built in server servers static files and doesn't allow for
+// any form of routing. This is a simple proxy server that forwards
+// requests to esbuild's server and returns the index.html if esbuild
+// returns a 404.
+export default function esbuildServerProxy(
```

Comment:
Yes, although I can't see exactly how Render achieves the same effect.

The problem is that if I open /examples/group-proof in a browser, and this issues a HTTP request, the server should simply return index.html and the JS for the single-page app. Without this, esbuild's server doesn't do it. I can't see any redirects/rewrites configured in render.com, but presumably something is implementing this since it does actually work.

Reply:
the way it works is that nextjs can export a website into a directory structure - so no redirects are necessary. you can see this locally by running `yarn export` in `consumer-client`

Reply:
Oh, I didn't notice that we're using the static `export` version.

In that case, I think we will have an issue with linking on Render. Is there a way to store redirect/rewrite configuration in the repository?